processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -263,4 +264,24 @@ public static Float convertObjectToFloat(Object valObj)
       throw new ParseException("Unknown type[%s]", valObj.getClass());
     }
   }
+
+  public static Long getIntegralFromFloatString(String floatStr)
+  {
+    BigDecimal convertedBD;
+    try {
+      convertedBD = new BigDecimal(floatStr);
+    } catch (NumberFormatException nfe) {
+      return null;
+    }
+
+    long asLong;
+    try {
+      asLong = convertedBD.longValueExact();
+    } catch (ArithmeticException ae) {
+      // indicates there was a non-integral part, or the BigDecimal was too big for a long
```

Comment:
I think there's also a bit of a consistency argument for allowing non-integral filter values for select/in.

I feel like the use case for non-integral values is somewhat stronger for Bound (e.g., comparing longs to a floating point average), but I also think it'd be a bit weird in terms of setting expectations to allow non-integral values in bound but generate an error if used in selector/in.